src/sleep/sleep.c

```diff
@@ -286,6 +286,11 @@ static int execute_s2h(const SleepConfig *sleep_config) {
                 last_capacity = r;
                 before_timestamp = now(CLOCK_BOOTTIME);
 
+                previous_discharge_rate = get_battery_discharge_rate();
+                if (previous_discharge_rate > 0)
+                        suspend_interval = ((last_capacity / previous_discharge_rate) * 60 - 30) * USEC_PER_MINUTE;
```

Comment:
This calculation could probably use a comment why we use these constants specifically.

Reply:
Sure